FilteredAnnouncementForUser

```diff
@@ -1,25 +1,14 @@
-        private static IEnumerable<AnnouncementModel> GetFilteredAnnouncementForUserUsingFilter2(IEnumerable<AnnouncementViewEntity> announcements,
-        int userID, List<int> docCatID, int noOfAnnouncement, List<string> postedDate, bool thumbnailVisibility)
+        private static IEnumerable<AnnouncementModel> GetFilteredAnnouncementForUserUsingFilter(IEnumerable<AnnouncementViewEntity> announcements,
+        int userId, List<int> filterByDocumentCategoryIds, List<string> filterByPostedDates, bool thumbnailVisibility)
         {
             IEnumerable<AnnouncementModel> FilteredAnnouncement = null;
-            LingMetaData metaData = new LingMetaData();
-
-            FilteredAnnouncement = (from ann in announcements
-                                    where (docCatID.Contains(ann.DocumentCategoryID ?? 0) && (ann.Posted.Date >= DateTime.Parse(postedDate[0]) && ann.Posted.Date <= DateTime.Parse(postedDate[1])))
-                                    select new AnnouncementModel
-                                    {
-                                        AnnouncementTypeID = ann.AnnouncementTypeID,
-                                        DocumentCategoryID = ann.DocumentCategoryID,
-                                        Label = ann.Posted.ToString("MM.dd.yyyy") + " - " + ann.Title,
-                                        EndTimeZoneInfo = ann.EndTimeZoneInfo,
-                                        StartTimeZoneInfo = ann.StartTimeZoneInfo,
-                                        AnnouncementAccessed = ann.AnnouncementAccessed,
-                                        CoverImageURL = AnnouncementManager.GenerateCoverImageURLFromCoverImageID(ann.CoverImageID),
-                                        AnnouncementIcon = AnnouncementManager.IconForAlert(ann.AnnouncementTypeID, ann.Filename),
-                                        AnnouncementPostedDate = ann.Posted.ToString("MMM yyyy"),
-                                        ThumbnailVisibility = thumbnailVisibility ? "inline-block" : "none",
-                                        Highlight = AnnouncementManager.HighLightAnnouncement(ann.Id) //update this to column highlight
-                                    }).Distinct();
+
+            FilteredAnnouncement = (
+                from announcement in announcements
+                where filterByDocumentCategoryIds.Contains(announcement.DocumentCategoryID ?? 0)
+                where announcement.Posted.Date >= DateTime.Parse(filterByPostedDates[0]) && announcement.Posted.Date <= DateTime.Parse(filterByPostedDates[1])
+                select AnnouncementModel.MapFrom(announcement, thumbnailVisibility)
+                ).Distinct();
 
             FilteredAnnouncement = FilteredAnnouncement.Select(annModel =>
```

Comment:
Removed since the properties referenced in this section aren't used and would result in null exceptions or not execute at all